### ISSUE TITLE
Add regendoc to CI?

### ISSUE THREAD

**Issue by The-Compiler** [pytest-dev/pytest#8808]
Looking at #8807 apparently we don't run regendoc on CI? It seems to me like we should, and not only run it on releases.

Perhaps we should even enforce that regendocs runs through without any diff, i.e. that any possible changes in the generated docs should be part of the PR diff?

**Comment by nicoddemus** [comment]
> Looking at #8807 apparently we don't run regendoc on CI? It seems to me like we should, and not only run it on releases.

👍 

> Perhaps we should even enforce that regendocs runs through without any diff, i.e. that any possible changes in the generated docs should be part of the PR diff?

If we can automate that it would be great, but I would be 👎 to force the author to do it; it requires Linux, it is sensitive to what's installed on the system, etc.

As a first step just running regendoc and ensure it doesn't error out would be a step in the right direction IMHO.

**Comment by The-Compiler** [comment]
Can we run it via pre-commit somehow, so that the diff ends up getting pushed as part of its auto-fixes?

**Comment by nicoddemus** [comment]
hmm probably, that's a good idea, however this will bring with it its own set of problems, because regendoc is sensitive to Python versions you have installed and the OS. :/

**Comment by RonnyPfannschmidt** [comment]
@asottile does pre-commit have all the python versions? (maybe via deadsnakes)

**Comment by RonnyPfannschmidt** [comment]
@nicoddemus i just realied that this would be ci only, as uses have different mileages

**Comment by nicoddemus** [comment]
Hmm good point, it also needs to run over all files, because it might capture changes in `src`, even thought it affects `doc` (pytest output for example).

**Comment by The-Compiler** [comment]
Yeah, I was thinking about adding a PR comment only showing the changes for that PR, but that doesn't seem possible. But catching issues revealed by regendoc early seems pretty beneficial, looking at how this release turned out :sweat_smile: 

So we could either run it on pre-commit or somesuch (or custom workflow pushing follow-up commits) and just tell people to not worry about it locally; or perhaps have an automated weekly PR with regendoc updates so we catch issues much earlier?

**Comment by RonnyPfannschmidt** [comment]
lets just have a auto suggestion mr action

so that when regendoc on master goes "bad" we get a pr directly as nxt thing for master

**Comment by nicoddemus** [comment]
> or custom workflow pushing follow-up commits

I think this would be the ideal solution, it would keep the related changes contained in the same PR plus make errors easy to track.

**Comment by asottile** [comment]
> 
> 
> @asottile does pre-commit have all the python versions? (maybe via deadsnakes)

I assume you mean pre-commit.ci?  these are the available versions: https://github.com/pre-commit-ci/runner-image/blob/info/versions.md#python
